modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EChain.java

```diff
@@ -233,6 +233,7 @@ private void analyzeWrite(final CompilerSettings settings, final Definition defi
         if (last instanceof ADefLink) {
             final ADefLink lastDef = (ADefLink) last;
             expression.analyze(settings, definition, variables);
+            // TODO: does it make more sense to just re-use last.after instead of using storeValueType?
```

Comment:
@uschindler No worries about the issue.  Definitely a somewhat strange corner case :)  I agree on keeping it as a marker.  It's very clean and clear way to identify these special cases.  Do you think it makes more sense to have it as an interface once storeValueType is removed?